[Fuzzer Binaries] Add libwebrtc fuzzers

JonWBedard opened the pull request:
#### 3a69dff897c192ea3567d5207f1787c2e035279d
<pre>
[Fuzzer Binaries] Add libwebrtc fuzzers
<a href="https://bugs.webkit.org/show_bug.cgi?id=283432">https://bugs.webkit.org/show_bug.cgi?id=283432</a>
<a href="https://rdar.apple.com/140288399">rdar://140288399</a>

Reviewed by David Kilzer.

This change requires a follow-up in <a href="https://bugs.webkit.org/show_bug.cgi?id=287154">https://bugs.webkit.org/show_bug.cgi?id=287154</a> to re-enable
the sdp_integration_fuzzer, which we disabled because of the gmock dependency.

* Source/ThirdParty/libwebrtc/Configurations/aom_av1_encoder_fuzzer.xcconfig: Install when ENABLE_LIBFUZZER is true.
* Source/ThirdParty/libwebrtc/Configurations/h264_bitstream_parser_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/h264_depacketizer_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/h265_bitstream_parser_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/h265_depacketizer_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_depacketizer_av1_assemble_frame_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_packetizer_av1_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_packetizer_h264_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_packetizer_h265_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_packetizer_vp8_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_packetizer_vp9_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/sdp_integration_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp8_dec_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp8_depacketizer_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp8_encoder_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp8_qp_parser_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp8_replay_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp9_dec_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp9_depacketizer_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp9_encoder_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp9_qp_parser_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp9_replay_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/webm_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj: Build fuzzers (except sdp_integration_fuzzer) as part of libwebrtc.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/xcshareddata/xcschemes/libwebrtc.xcscheme: Rename libwebrtc to libwebrtc.dylib so
that libwebrtc can target fuzzers too.

Canonical link: <a href="https://commits.webkit.org/289949@main">https://commits.webkit.org/289949@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f5d9568dd1c9fba556c278c6cefdb5835138614d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88490 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8009 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42930 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93448 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39244 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8396 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16193 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/68253 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25968 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91492 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6425 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80040 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48620 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6197 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/34456 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38352 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76570 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/35339 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95290 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15665 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/11484 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/77116 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15921 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75896 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76379 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18806 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20771 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/19187 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/8704 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15681 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15422 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18871 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17204 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer reviewed (changes requested):
Need to understand why `sdp_integration_fuzzer` broke before we land this.

I'm looking at that now.

ddkilzer commented on Source/ThirdParty/libwebrtc/Configurations/h264_bitstream_parser_fuzzer.xcconfig:
```diff
@@ -23,6 +23,13 @@
 
 #include "Base-libwebrtc.xcconfig"
 
+EXCLUDED_SOURCE_FILE_NAMES = $(EXCLUDED_SOURCE_FILE_NAMES_$(ENABLE_LIBFUZZER));
+EXCLUDED_SOURCE_FILE_NAMES_ = *;
+EXCLUDED_SOURCE_FILE_NAMES_YES = ;
```
Why are these needed in some targets, but not (for example) `aom_av1_encoder_fuzzer.xcconfig`?

Or maybe `aom_av1_encoder_fuzzer.xcconfig` just needs this added?  :)

JonWBedard replied:
These lines should be added for everything, I missed some of the fuzzers you just added in https://commits.webkit.org/289734@main

ddkilzer commented on Source/ThirdParty/libwebrtc/Configurations/vp8_encoder_fuzzer.xcconfig:
```diff
@@ -26,11 +26,13 @@
 OTHER_LDFLAGS = $(inherited) -lwebrtc;
 
 WARNING_CFLAGS = $(inherited) -Wno-constant-logical-operand;
+
```
Do we need to add these lines here?

```suggestion

EXCLUDED_SOURCE_FILE_NAMES = $(EXCLUDED_SOURCE_FILE_NAMES_$(ENABLE_LIBFUZZER));
EXCLUDED_SOURCE_FILE_NAMES_ = *;
EXCLUDED_SOURCE_FILE_NAMES_YES = ;

```

ddkilzer commented on Source/ThirdParty/libwebrtc/Configurations/vp9_encoder_fuzzer.xcconfig:
```diff
@@ -26,11 +26,13 @@
 OTHER_LDFLAGS = $(inherited) -lwebrtc;
 
 WARNING_CFLAGS = $(inherited) -Wno-constant-logical-operand;
+
```
Do we need to add these lines here?

```suggestion

EXCLUDED_SOURCE_FILE_NAMES = $(EXCLUDED_SOURCE_FILE_NAMES_$(ENABLE_LIBFUZZER));
EXCLUDED_SOURCE_FILE_NAMES_ = *;
EXCLUDED_SOURCE_FILE_NAMES_YES = ;

```

JonWBedard replied:
We do, this is an oversight. (actually somewhat surprised this built in EWS without these lines)

ddkilzer commented on Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
```diff
@@ -27278,19 +27312,14 @@
 			target = 449467C22AE05DA200A9FED0 /* rtp_depacketizer_av1_assemble_frame_fuzzer */;
 			targetProxy = 449467D22AE05DD200A9FED0 /* PBXContainerItemProxy */;
 		};
-		449CF1612ADEDE9B00F22CAF /* PBXTargetDependency */ = {
-			isa = PBXTargetDependency;
-			target = 44ABBE882AC63F0C006B44DD /* sdp_integration_fuzzer */;
-			targetProxy = 449CF1602ADEDE9B00F22CAF /* PBXContainerItemProxy */;
-		};
```
Is there a way we can be clever and disable building `sdp_integration_fuzzer` when building with automation, but leave `sdp_integration_fuzzer` enabled when building at desk for engineers?

Maybe this isn't a common enough use case to do this, though.

We should have a bug or radar tracking the fix, though.  (We probably do, but it's late, and I'm too tired to look it up.)

JonWBedard replied:
I think this is not common enough to do. My hope is also that this is short-lived.

Tracked in https://bugs.webkit.org/show_bug.cgi?id=287154, which I'll be mentioning in the new commit message.

ddkilzer reviewed (approved):
r=me, but please make sure we have a bug/radar tracking tracking the fix for `sdp_integration_fuzzer`.

Also, make sure we add the `EXCLUDED_SOURCE_FILE_NAMES` macros as needed for other `*.xcconfig` files.

JonWBedard committed:
[Fuzzer Binaries] Add libwebrtc fuzzers
https://bugs.webkit.org/show_bug.cgi?id=283432
rdar://140288399

Reviewed by David Kilzer.

This change requires a follow-up in https://bugs.webkit.org/show_bug.cgi?id=287154 to re-enable
the sdp_integration_fuzzer, which we disabled because of the gmock dependency.

* Source/ThirdParty/libwebrtc/Configurations/aom_av1_encoder_fuzzer.xcconfig: Install when ENABLE_LIBFUZZER is true.
* Source/ThirdParty/libwebrtc/Configurations/h264_bitstream_parser_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/h264_depacketizer_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/h265_bitstream_parser_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/h265_depacketizer_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_depacketizer_av1_assemble_frame_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_packetizer_av1_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_packetizer_h264_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_packetizer_h265_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_packetizer_vp8_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/rtp_packetizer_vp9_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/sdp_integration_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp8_dec_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp8_depacketizer_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp8_encoder_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp8_qp_parser_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp8_replay_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp9_dec_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp9_depacketizer_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp9_encoder_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp9_qp_parser_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/vp9_replay_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/Configurations/webm_fuzzer.xcconfig: Ditto.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj: Build fuzzers (except sdp_integration_fuzzer) as part of libwebrtc.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/xcshareddata/xcschemes/libwebrtc.xcscheme: Rename libwebrtc to libwebrtc.dylib so
that libwebrtc can target fuzzers too.

Canonical link: https://commits.webkit.org/289949@main
